### PR TITLE
Wrap supplementary node blocks to enable resizing them.

### PR DESCRIPTION
**Pull request by wiseoldduck**
Most ASCellNodeBlocks are wrapped by ASCollectionView.mm to add an `interactionDelegate` as well as calling `enterHierarchyState` on the node and setting its transform to a reflection if `inverted` is set on the collectionView.

This PR adds this behavior to supplementary nodes as well.

### CODE REVIEW

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/4c68885e5b11f002c65c3a0eab0be376/raw/68e71083051d97f11a6fe7733d03be841fa2808a/log.txt)

File: Source/ASCollectionView.mm [hunk 1]
```diff
@@ -2015,7 +2015,26 @@ - (ASCellNodeBlock)dataController:(ASDataController *)dataController supplementa
     block = ^{ return cell; };
   }
 
-  return block;
+  // Wrap the node block
+  BOOL disableRangeController = ASCellLayoutModeIncludes(ASCellLayoutModeDisableRangeController);
```
Comment (maicki) [inline Source/ASCollectionView.mm]: We may should remove that ;)

File: Source/ASCollectionView.mm [hunk 2]
```diff
@@ -2015,7 +2015,22 @@ - (ASCellNodeBlock)dataController:(ASDataController *)dataController supplementa
     block = ^{ return cell; };
   }
 
-  return block;
+  // Wrap the node block
+  __weak __typeof__(self) weakSelf = self;
```
Comment (nguyenhuy) [inline Source/ASCollectionView.mm]: At one point we should add strongify & weakify macros.

File: Source/ASCollectionView.mm [hunk 3]
```diff
@@ -2015,7 +2015,22 @@ - (ASCellNodeBlock)dataController:(ASDataController *)dataController supplementa
...
+  return ^{
+    __typeof__(self) strongSelf = weakSelf;
+    ASCellNode *node = (block ? block() : cell);
```
Comment (nguyenhuy) [inline Source/ASCollectionView.mm]: Isn't block is guaranteed to be non-nil after [this line](https://github.com/TextureGroup/Texture/pull/1265/files#diff-8b82b1556d1293f6f9789beb236c9271R2015)? Maybe remove that line to be consistent with the implementation of `dataController:nodeBlockAtIndexPath:`.

Comment (maicki) [reply Source/ASCollectionView.mm]: Concur with @nguyenhuy. The block should not be nil in here and the returning block is retaining the block too.

Comment (wiseoldduck) [reply Source/ASCollectionView.mm]: I thought you had to `copy` a block to actually retain it

Comment (wiseoldduck) [reply Source/ASCollectionView.mm]: nm I'm like stuck in 2010 😛

**Pull request merged**